app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def increase_user_counter
+    User.find_by(id: user.id).update(post_counter: user.post_counter.nil? ? 1 : ++user.post_counter)

```

Comment:
- [ ] if you set the foreign key in the `User` model and set the belongs to in this model to the foreign and class_name: user the following can look like below ```  author.increment!(:posts_counter)```
- [ ] [OPTIONAL]after that you can also add something like `after_save :update_post_counter` and make this method private